src/btree/bt_curnext.c

```diff
@@ -172,7 +174,17 @@ __cursor_var_append_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_
             return (WT_NOTFOUND);
 
         __cursor_set_recno(cbt, WT_INSERT_RECNO(cbt->ins));
+
 restart_read:
+        /*
```

Comment:
We're going to need testing around this restart_read edgecase.

Reply:
Can be done in another ticket if required.

Reply:
This will be done in WT-9475, with writing the prepared transactions test